.github/workflows/pdm-lock-automation.yaml

```diff
@@ -76,17 +92,41 @@ jobs:
           else
             echo "Virtual environment already exists."
           fi
+
           source .venv/bin/activate
+
+          echo "Generating/Updating pdm.lock..."
           pdm lock -G :all -v
+
+          echo "Stashing changes if there are any..."
+          git stash || echo "No stash entries found, continuing..."
```

Comment:
@kirtimanmishrazipstack Also just curious on what kind of uncommitted changes would be present if at all, as this is running inside the workflow env.

Reply:
@hari-kuriakose if the developer running the automation have some uncommitted changes in any of files, including manually generating pdm lock. This will handle all the scenarios

Reply:
@kirtimanmishrazipstack there won't be any uncommitted files other than pdm lock files.

Reply:
@ritwik-g I meant that, what if the developer has pushed only pyproject file, but he/she is working on some other files. 
When the workflow runs, all the files the user has changed but not committed will come as uncommitted right?